src/org/zaproxy/zap/extension/ascanrulesBeta/UsernameEnumeration.java

```diff
@@ -139,11 +139,18 @@ public void scan() {
 		try {
 			boolean loginUrl = false;
 
-			// Are we dealing with a login url in any of the contexts of which this uri is part
-			URI requestUri = getBaseMsg().getRequestHeader().getURI();
 			ExtensionAuthentication extAuth = (ExtensionAuthentication) Control.getSingleton()
 					.getExtensionLoader().getExtension(ExtensionAuthentication.NAME);
 
+			if(extAuth.getModel().getSession().getContexts().isEmpty()) {
```

Comment:
ZAP has a context by default so in most cases this check will be false. I think we should also check that at least one of the existing contexts have an authentication method (in this case FormBasedAuthenticationMethod) before continuing.